hibernate-core/src/test/java/org/hibernate/optional/javassist/OptionalJavassistDependencyTest.java

```diff
@@ -0,0 +1,26 @@
+/*
+ * Hibernate, Relational Persistence for Idiomatic Java
+ *
+ * License: GNU Lesser General Public License (LGPL), version 2.1 or later
+ * See the lgpl.txt file in the root directory or http://www.gnu.org/licenses/lgpl-2.1.html
+ */
+package org.hibernate.optional.javassist;
+
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.ExpectedException;
+
+import static org.junit.Assert.assertNull;
+
+public class OptionalJavassistDependencyTest {
+
+	@Rule
+	public ExpectedException thrown = ExpectedException.none();
+
+	@Test
+	public void shouldNotBeOnTheClasspath() throws Exception {
+		thrown.expect( ClassNotFoundException.class );
+
+		Class<?> aClass = getClass().getClassLoader().loadClass( "javassit.Classpath" );
```

Comment:
this test will always succeed, even when javassist is on classpath as you have a typo in the package name :)